src/swapd/swap_state.rs

```diff
@@ -1374,7 +1374,7 @@ fn try_bob_cancel_final_to_swap_end(
             // send swap outcome to farcasterd
             Ok(Some(SwapStateMachine::SwapEnd(Outcome::FailureRefund)))
         }
-        _ => Ok(None),
+        _ => try_punish_to_swap_end(event, runtime),
```

Comment:
I'm not seeing where Bob is watching the Punish transaction. Does that still need to be added?

Reply:
it's working now on c1bcec836ddda10ce3ccd5038bff1a91f8dfe962 - I've added a description to this PR.

I've obviously merged in your work from #855 that enables filtering address transactions for IO direction, but also merged #863 to avoid unnecessary rewrites since it touches related code. My branch here is currently a mess, so will merge in both of yours first and then rebase & clean up this one so it can be reviewed.

Reply:
didn't rebase since it introduced some conflicts & created a broken history from overlapping changes with #863, but I've squashed commits where relevant to produce a reviewable history - assuming you skip all the merges.